lib/rules/default-param-last.js

```diff
@@ -5,6 +5,35 @@
 
 "use strict";
 
+/**
+ * checks if node is optional parameter
+ * @param {ASTNode} node the node to be evaluated
+ * @returns {boolean} whether the node is optional
+ */
+function isOptionalParam(node) {
+    return (
+        (node.type === "ArrayPattern" ||
+            node.type === "AssignmentPattern" ||
+            node.type === "Identifier" ||
+            node.type === "ObjectPattern" ||
+            node.type === "RestElement") &&
+        node.optional
+    );
+}
+
+/**
+ * checks if node is plain parameter
+ * @param {ASTNode} node the node to be evaluated
+ * @returns {boolean} whether the node is plain
+ */
+function isPlainParam(node) {
+    return !(
+        node.type === "AssignmentPattern" ||
+        node.type === "RestElement" ||
+        isOptionalParam(node)
```

Comment:
`isOptionalParam` is also checking for `"RestElement"` and `"AssignmentPattern"`. It seems like there should be a way to avoid checking this twice when `isPlainParam()` is called.